### ISSUE TITLE
Keyword arguments not used when resolving its documentation and tags

### ISSUE THREAD

**Issue by stdedos** [robotframework/robotframework#3238]
I have this test:
```robot
*** Settings ***

*** Variables ***
${test}   foobar
&{dict}   foo=bar   bar=baz   baz=foo

*** Test Cases ***
Test Kwargs
  ${my_user} =        Set Variable  damn
  Log My User         ${my_user}  ${test}
  Log My Other User   ${my_user}  ${test}

*** Keywords ***
Log My User
  [Documentation]   Log ${user} doing ${test}
  [Arguments]       ${user}   ${test}
  Log Many          ${user}   ${test}

Log My Other User
  [Documentation]   Log ${my_user} doing ${test}
  [Arguments]       ${my_user}  ${test}
  Log Many          ${my_user}  ${test}
```

and I see:

![image](https://user-images.githubusercontent.com/133706/61043480-0a753a00-a3df-11e9-9d39-7c5c6f8bb0d8.png)

I would expect that keyword's arguments are taken into account when resolving variables, if all pre-defined variables are taken into account too.

**Comment by pekkaklarck** [comment]
Good catch. Same problem also with keyword tags. Ought to be easy to fix.

**Comment by pekkaklarck** [comment]
I was wrong, this is surprisingly complicated to fix.

The problem is that if we wait until arguments have been added to variables before we resolve doc and tags, we have already called an internal `start_keyword` method that logs doc and tags to output.xml. We could move logging doc and tags to `end_keyword`, but that breaks `--flattenkeywords` because it must found tags before executed keywords in output.xml. This approach probably would work with doc, though.

I guess this could be handled by adding a new method like `kw_info` to the XML logger to log kw doc and tags. It could be called after `start_keyword` but before starting to execute internal keywords. Could also see could calling `start_keyword` be delayed, but that may be impossible due to error handling.

**Comment by pekkaklarck** [comment]
Too much work compared to benefits to be included in RF 3.2.

**Comment by VernonCrabtree** [comment]
Having tried to work on the old lexer, I highly appreciate the work done and was enthusiastic about taking another look at the logging control (particularly due to enforced holidays due to Covid-19).
We had discussed using tags for logging control and it occurred to me that this could, in turn, be controlled via arguments. However, I stumbled on this issue with this approach.
It is a pitty a kw_info method has not been implemented - this could be very helpful for logging control in the future.

One thing I did notice: arguments ARE evaluated within Tags, but within the calling context. This is context leakage, which I would consider to be fairly dangerous.

**Comment by pekkaklarck** [comment]
You can test does the change have adverse effects by running all acceptance tests. If they pass, the change ought to be pretty safe.

Notice also that one big theme with the next RF version will be enhancing the execution side in general.

**Comment by VernonCrabtree** [comment]
OK - I am learning more about these tests. They failed.
I'll get the tests to pass and see about a merge request after that . . .

**Comment by VernonCrabtree** [comment]
Pull request #3566 has passed all tests - even the 2.7 ones.
